### PR TITLE
Snippet Ordering

### PR DESCRIPTION
**Pull request by AlexJerabek**
The big change here is reordering the snippets to have a more logical display. The top level folder look like this:
- Basics
- Scenarios
- Critical objects (range, paragraph, etc in alphabetical order)
- Non-critical objects (chart, PivotTable, etc in alphabetical order)
- Code patterns
- Fun stuff

Within a folder, the snippets are organized alphabetically by title. I edited the Excel snippet titles and descriptions to match the format now used in the Word snippets.

I also removed the following Excel snippets because they were not unique (placing useful code in other snippets).
- Copy and multiply values (range)
- Create and use an intersection of ranges (range)
- Create named item (named items)

As always, I'm open to suggestions and feedback. Thanks!

### CODE REVIEW

**Comment by kbrandl** [comment]
@AlexJerabek I'm not ignoring this PR, just haven't had time today to give it a good review; I'll get you some feedback tomorrow (Friday) morning. Apologies for the delay!

**Comment by AlexJerabek** [comment]
@kbrandl I wouldn't blame you if you were ignoring it ;-)

File: samples/excel/46-named-item/list-named-items.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
-order: 4
+order: 2
 id: excel-named-item-list-named-items
-name: List all named items in a workbook
-description: List all named items in a workbook
+name: List all
+description: Lists all the named items in a workbook.
```
Comment (kbrandl) [inline samples/excel/46-named-item/list-named-items.yaml]: Suggest changing **description** -> `Lists all named items in a workbook.`

File: samples/excel/70-code-patterns/multiple-property-set.yaml [hunk 2]
```diff
@@ -1,6 +1,7 @@
-id: excel-multiple-property-set
-name: Multiple Property Set
-description: Setting multiple properties at once with the rich API object set() method.
+order: 1
+id: excel-code-patterns-multiple-property-set
+name: Set multiple properties at once
```
Comment (kbrandl) [inline samples/excel/70-code-patterns/multiple-property-set.yaml]: Suggest changing **name** -> `Set multiple properties`

**Comment by AlexJerabek** [comment]
@kbrandl I'm realizing that renaming and reordering the folders will break any links to these snippets. Is that a concern? Do we every link directly to snippets? Or have we avoided that for just this reason?

**Comment by kbrandl** [comment]
@AlexJerabek re your question about impact of reorganizing snippets -- I'm not aware of anywhere where we link directly to individual snippets. 

Anytime you change the snippet ID (in the snippet yaml file) or a method name within a snippet, you should update the corresponding mapping file (in the `snippet-extractor-metadata` folder) if necessary -- but I don't think simply reorganizing the structure of snippets should have any implications here.

**Comment by AlexJerabek** [comment]
Thanks @kbrandl! I've got a new version up. Based on our in-person conversation, I changed the ordering structure to be an alphabetical list of objects. I merged the "Basics", "Code Patterns", and "Scenarios" folders into "Guidance". My logic is those snippets aren't samples of APIs, rather our advice to developers interesting in learning more.

File: samples/excel/14-conditional-formatting/conditional-formatting-basic.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
-order: 1
+order: 2
 id: excel-range-conditional-formatting-basic
-name: Conditional Formatting for Ranges - Basic
-description: Apply common types of conditional formatting to ranges.
+name: Basic onditional formatting
```
Comment (kbrandl) [inline samples/excel/14-conditional-formatting/conditional-formatting-basic.yaml]: typo: missing `c` at the start of `conditional`

File: samples/excel/30-events/setting-changed.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
-order: 3
+order: 8
 id: excel-events-setting-changed
-name: Handle the settings-changed event
-description: This snippet shows how to register a handler for the SettingsChanged event.
+name: Workbook events
+description: Registers an event handler that runs when workbook's settings change.
```
Comment (kbrandl) [inline samples/excel/30-events/setting-changed.yaml]: either add the word "the" -> `...when the workbook's settings change.`   OR remove `'s` -> `...when workbook settings change.`

File: samples/excel/30-events/events-worksheet-selectionchanged.yaml [hunk 5]
```diff
@@ -1,7 +1,7 @@
-order: 8
+order: 13
 id: excel-events-worksheet-selectionchanged
-name: Events - Worksheet onSelectionChanged
-description: Add an event handler for the worksheet onSelectionChanged event
+name: Worksheet selection change
```
Comment (kbrandl) [inline samples/excel/30-events/events-worksheet-selectionchanged.yaml]: for consistency, add the word 'event' -> `Worksheet selection change event`

File: samples/excel/54-worksheet/add-delete-rename-move-worksheet.yaml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 order: 2
 id: excel-worksheet-add-delete-rename-move-worksheet
-name: 'Add, delete, rename and move worksheet'
-description: 'Add, delete, rename and change the position of a worksheet'
+name: 'Add, delete, rename, and move worksheet'''
```
Comment (kbrandl) [inline samples/excel/54-worksheet/add-delete-rename-move-worksheet.yaml]: remove 2 extra quote marks from the end of the **name**

File: samples/excel/54-worksheet/add-delete-rename-move-worksheet.yaml [hunk 7]
```diff
@@ -1,7 +1,7 @@
...
+description: 'Adds, deletes, renames, and moves a worksheet.'''
```
Comment (kbrandl) [inline samples/excel/54-worksheet/add-delete-rename-move-worksheet.yaml]: remove 2 extra quote marks from the end of the **description**

**Comment by AlexJerabek** [comment]
Thank again @kbrandl! Updated with all your feedback.

**Pull request merged**